### PR TITLE
fix: escape regex when square brackets are present

### PR DESCRIPTION
**Pull request by dorobila**
Stumbled across react-highlight-words library and noticed that when I add "[" character, it breaks the whole app.

This PR should fix both packages.

image
image

### CODE REVIEW

File: src/utils.test.js [hunk 1]
```diff
@@ -66,6 +66,17 @@ describe('utils', () => {
...
+  it('should handle unclosed square brackets when autoEscape prop is truthy', () => {
+    const rawChunks = Chunks.findChunks({
+      autoEscape: true,
+      searchWords: ['text['],
+      textToHighlight: ']This is text['
+    })
+    expect(rawChunks).to.eql([
+      {start: 9, end: 14, highlight: false}
+    ])
+  })
```
Comment (bvaughn) [inline src/utils.test.js]: This test also passes on the main branch without the accompanied regex change. Is there a different test that we could add that captures the original bug?

Comment (dorobila) [reply src/utils.test.js]: hmm, I'll look into it

**Comment by dorobila** [comment]
I apologize for my previous commit. Upon further investigation while I was writing a test for it, I discovered that I was using the functions incorrectly andv I've incorrectly assumed that the autoEscape property automatically escapes special characters by default.

I have now reverted my changes and set the autoEscape property to true, as I believe this should be the default behavior.

Let me know what you think.

**Review by bvaughn — COMMENTED** [review]
Hmm. Changing that default value has performance implications. I'm not sure how I feel about it. (To be fair, I haven't used this library in a long time and so I feel less sure about judging the impact of any change.)

**Comment by dorobila** [comment]
Fair enough. Let's just hope no one mistypes "[" instead of "p" without setting autoEscape becasue I certainly did and it broke something similar that I wrote that it highlights some text when searching for something😄.

Feel free to close the PR.
Thank you for the chat.

**Comment by bvaughn** [comment]
We could certainly update the demo website :)

**Comment by dorobila** [comment]
Sounds good. I'll create a PR on [react-highlight-words](https://github.com/bvaughn/react-highlight-words) demo's website so that it autoescapes by default.
Thought about adding a checkbox similar to "Case Sensitive?", but if autoEscape is ticked off, it'll break the app and users are going to have to refresh the page.
The example under Usage and the codesandbox are using autoEscape by default, so most of the time users are going to copy/pase the example and build from there.

image

**Pull request closed without merging**